src/common.jl

```diff
@@ -298,6 +300,47 @@ function Base.iszero(p::AbstractPolynomial)
     return all(iszero.(coeffs(p))) && p[0] == 0
 end
 
+# See discussions in https://github.com/JuliaMath/Polynomials.jl/issues/258
+is_(fn, p::AbstractPolynomial) = all(fn, coeffs(p))  # Do not export
+as_(fn, p::P) where {P<:AbstractPolynomial} = ⟒(P)(fn.(coeffs(p)), p.var)  # Do not export
+
+"""
+    isreal(p::AbstractPolynomial)
+
+Determine whether a polynomial is a real polynomial, i.e., having only real numbers as coefficients.
+
+See also: [`real`](@ref)
+"""
+Base.isreal(p::AbstractPolynomial) = is_(isreal, p)
+"""
+    real(p::AbstractPolynomial)
+
+Construct a real polynomial from the real parts of the coefficients of `p`.
+
+See also: [`isreal`](@ref)
+
+!!! note
+    This could cause losing terms in `p`. This method is usually called on polynomials like `p = Polynomial([1, 2 + 0im, 3.0, 4.0 + 0.0im])` where you want to chop the imaginary parts of the coefficients of `p`.
+"""
+Base.real(p::AbstractPolynomial) = as_(real, p)
+
+"""
+    isintegerpoly(p::AbstractPolynomial)
+
+Determine whether a polynomial is an integer polynomial, i.e., having only integers as coefficients.
```

Comment:
Maybe comment "as determined by `isinteger`" (so that readers know how this is determined

Reply:
Do we really need that? "having only integers as coefficients" is kind of straightforwardly implying `isinteger`. And when we publish the docs, users can just click the "source" button to see how it is implemented.